src/main/java/org/cactoos/text/TextOfIterator.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2020 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.text;
+
+import java.util.Iterator;
+
+/**
+ * Text of {@link Iterator}
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @since 0.49
+ */
+public final class TextOfIterator extends TextEnvelope {
+
+    /**
+     * Ctor.
+     *
+     * @param iterator The iterable to convert to string
+     */
+    public TextOfIterator(final Iterator<Character> iterator) {
+        super(
+            new TextOfScalar(
+                () -> {
+                    final StringBuilder buf = new StringBuilder();
+                    iterator.forEachRemaining(buf::append);
+                    return buf.toString();
+                }
```

Comment:
@baudoliver7 this should instead delegate to `new TextOfIterable(new IterableOf(iterator))`, also I think we can just take an `Iterator<?>`